tests/Feature/BookTest.php

```diff
@@ -114,6 +114,39 @@ public function test_getAllBooks_genre_error(): void
             ->assertInvalid(['genre']);
     }
 
+    public function test_getAllBooks_search(): void
+    {
+        Book::factory()->count(1)->create(['title' => 'enim ']);
+
+        $response = $this->get('/api/books?search=enim');
+
+        $response->assertStatus(200)
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['data', 'message'])
+                    ->has('data', 1, function (AssertableJson $json) {
+                        $json->hasAll([
+                            'id',
+                            'title',
+                            'author',
+                            'image',
+                            'genre'
+                        ])->has('genre', function (AssertableJson $json) {
+                            $json->hasAll(['id', 'name']);
+                        });
+                    });
+            });
+    }
+
+    public function test_getAllBooks_search_failed(): void
+    {
+        Book::factory()->count(10)->create();
+        $data = null;
+        $response = $this->get("/api/books?search={$data}");
+
+        $response->assertStatus(302)
+            ->assertInvalid(['search']);
+    }
+
```

Comment:
You should add an extra test that creates a few books, then searches for something that doesn't exist. The test should confirm that data is empty in that case